ReactiveCocoa/Swift/Lifetime.swift

```diff
@@ -1,21 +1,41 @@
 import enum Result.NoError
 
-/// Provides a signal that completes when the lifetime object is deinitialized.
-///
-/// When assigned to a property of another object, provides a hook to
-/// observe when that object goes out of scope.
+/// Represent the lifetime of an object, and provide a hook to observe when the
+/// object deinitializes.
 public final class Lifetime {
 	/// A signal that sends a Completed event when the lifetime ends.
 	public let ended: Signal<(), NoError>
 
-	private let endedObserver: Signal<(), NoError>.Observer
-
-	public init() {
-		(ended, endedObserver) = Signal.pipe()
+	public init(_ token: Token) {
```

Comment:
Can you document this? In particular I would mention that `Lifetime` does not hold a strong reference to `Token`, as that would defeat the purpose.